app/controllers/users_controller.rb

```diff
@@ -1,29 +1,76 @@
 class UsersController < ApplicationController
+  before_action :logged_in_user, only: %i(edit update destroy)
+  before_action :correct_user, only: %i(edit update)
+  before_action :admin_user, only: :destroy
+  before_action :find_user, only: :show
```

Comment:
sắp xếp lại thứ tự 
```ruby
:logged_in_user
:find_user
:correct_user
:admin_user
```
bạn tìm hiểu tại sao lại như vậy nhé?

Reply:
đâu tiên xác nhận log in trước một số action (edit update destroy)
find_user để tìm user để hiển thị (show)
sau đó xác nhận đúng user mới có thể edit update tải khoán của mình
- admin có thể delete các user khác